### DISCUSSION TITLE
How should I label my own dataset to use docTR training scripts?

### DISCUSSION THREAD

**Discussion by ghost** [mindee/doctr#585] [Q&A] [answered]
I want to annotate my own dataset How can I annotate? which tool is used to annotate to run this code?

**Comment by charlesmindee** [comment, marked as answer]
Hi @hammadyounas2008, 

To  annotate your data you have several options:

- You can do it yourself (or get it done by a data labelling company) via a platform. Usually, you just upload your images, and the platform is designed to allow you to quickly circle/select areas/boxes on the image, and some tools even include OCR to pre-annotate words, so that you just have to validate or edit them. It is the best way to annotate data since it is human-sourced and then (supposed to be) very reliable. The cons: is it expensive in time and/or money.

- You have a cheeper/faster option: use OCR services to fully annotate your data (without human verification), but the quality of your data will obviously be limited by the robustness of the OCR used. You can use some tricks such as crossing the predictions of multiple OCRs on your dataset to increase the precision of the labelling.

Does it answer your question ?

**Reply by ghost** [reply]
@charlesmindee  Thanks for your reply. I have used VGG annotater tool online but the data is saved in another format, not like your format for training. Do you have any link where I can label or any website which labels my dataset automatically

**Reply by ghost** [reply]
@fg-mindee  Can you Reply to this too.?

**Reply by charlesmindee** [reply]
What is the output format of the VGG annotator ? I think we can find a way to convert these annotations to our format. When you label your data on a platform, you always need to convert the annotations to the desired format since there is no universal format, except if you design your own platform and set your own export format.

**Reply by ghost** [reply]
@charlesmindee The output format of VGG anotator is this  {"passbook_64_0.jpeg167515":{"filename":"passbook_64_0.jpeg","size":167515,"regions":[{"shape_attributes":{"name":"polygon","all_points_x":[23,23,176,181],"all_points_y":[85,100,104,86]},"region_attributes":{"name":"not_defined","type":"unknown","image_quality":{"good":true,"frontal":true,"good_illumination":true}}}],"file_attributes":{"caption":"","public_domain":"no","image_url":""}}}

**Reply by ghost** [reply]
How do you label the data to train the model?

**Reply by fg-mindee** [reply]
Hi @hammadyounas2008 :wave: 

> @fg-mindee Can you Reply to this too.?

Once again, this is an open source project and I feel like it would be good not to require two maintainers on a single question unless it's critical for the project :pray:

> How do you label the data to train the model?

Please check the already available resources for each task training (https://github.com/mindee/doctr/blob/main/references/classification/README.md, https://github.com/mindee/doctr/blob/main/references/detection/README.md, https://github.com/mindee/doctr/blob/main/references/recognition/README.md). That should give you all the information that you need about data format (you will have to handle the format conversion on your own).

More generally speaking, I suggest considering the scope of this project, which is not related in any way to data labeling. We provide resources, but those are limited. So please make the effort to check the project scope, whether your topic has already been discussed in open/closed issues & discussions before asking here :pray:  If you wish to write a tutorial on how to label the data, feel free to submit a PR :)
Open source is about sharing knowledge, it goes both ways when it's at its best :ok_hand: 

Have a nice day!

**Reply by rajsabi** [reply]
Hi @fg-mindee ! Hope you are fine. Can you please tell me how much data is required to train detection and recognition model for oriya language from scratch? I have 2M word images for the same.
